### PR TITLE
Configurable UDP and TCP Events + cURL API Calls

### PR DESCRIPTION
**Pull request by folkhack**
Hello there!

I am a very happy user of the PHP Datadog library that you've got here =)

Currently I am on a project where we need to utilize the UDP events that are outlined here: http://docs.datadoghq.com/guides/dogstatsd/#events

To utilize this I added the ability to send events via both UDP and TCP since I feel both are highly valuable in different scenarios (my companies use both). I have made UDP the default here so-as to just work "out of box" without needing an API configuration. If you call the "configure" method it is assumed you are utilizing the API. There are also a couple of cURL SSL configuration options that are important if folks are using self-signed certificates with reverse-proxies.

I have also greatly improved the API events call by swapping out the "file_get_contents" strategy with a cURL one. This checks for things like response codes, empty bodies, malformed JSON, etc. to get down to checking the "status" field passed back by Datadog. If we want verification that our event went out (via TCP) we may as well ensure that the web request was properly handled =)

I have kept in-line with your development practices to the best of my ability. If you wish to swap the default method for event communication from UDP to TCP let me know and I will re-submit a pull-request with that change.

Otherwise thank you for your time in reviewing these improvements! They will make a HUGE difference in our adoption of this library =)

Sincerely,
- John Parks

### CODE REVIEW

**Comment by folkhack** [comment]
This branch has been updated to include the recent UDP port change as well as adding StatsD hostname/port options to the "configure" method.

**Comment by hkaj** [comment]
@folkhack thanks a lot for this neat contribution, and apologies for the long wait. We'll review this soon, and it looks very promising :)

File: libraries/datadogstatsd.php [hunk 1]
```diff
@@ -239,28 +282,98 @@ public static function event($title, $vals = array()) {
...
+    private static function eventUdp($vals) {
+
+        // Format required values title and text
+        $title = isset($vals['title']) ? (string) $vals['title'] : '';
+        $text = isset($vals['text']) ? (string) $vals['text'] : '';
+
+        // Format fields into string that follows Datadog event submission via UDP standards
+        //   http://docs.datadoghq.com/guides/dogstatsd/#events
+        $fields = '';
+        $fields .= ($title) ? $title : '';
```
Comment (hkaj) [inline libraries/datadogstatsd.php]: is this needed? I believe L357 does that already

File: libraries/datadogstatsd.php [hunk 2]
```diff
@@ -239,28 +282,98 @@ public static function event($title, $vals = array()) {
...
+         */
+        $success = true;
 
         // Get the url to POST to
         $url = self::$__datadogHost . self::$__eventUrl
              . '?api_key='          . self::$__apiKey
              . '&application_key='  . self::$__applicationKey;
 
-        // Send, suppressing and logging any http errors
+        $curl = curl_init($url);
```
Comment (hkaj) [inline libraries/datadogstatsd.php]: Should we add a lib-curl requirement in composer to be safe here?

File: libraries/datadogstatsd.php [hunk 3]
```diff
@@ -239,28 +282,98 @@ public static function event($title, $vals = array()) {
...
+        $response_body = curl_exec($curl);
+        $response_code = (int) curl_getinfo($curl, CURLINFO_HTTP_CODE);
+
         try {
-            file_get_contents($url, 0, $context);
-        } catch (Exception $ex) {
-            error_log($ex);
+
+            // Check for cURL errors
+            if ($curlErrorNum = curl_errno($curl)) {
```
Comment (hkaj) [inline libraries/datadogstatsd.php]: Won't this always be true because it's an affectation?

File: libraries/datadogstatsd.php [hunk 4]
```diff
@@ -1,11 +1,17 @@
 <?php
 /**
  * Datadog implementation of StatsD
- * Added the ability to Tag!
+ * - Added the ability to Tag!
+ * - Most of this code was stolen from: https://gist.github.com/1065177/5f7debc212724111f9f500733c626416f9f54ee6
+ * - I did make it the most efficient UDP process possible, and add tagging.
  *
- * Most of this code was stolen from: https://gist.github.com/1065177/5f7debc212724111f9f500733c626416f9f54ee6
+ * UDP events + cURL implementation (following notes to be moved to CHANGELOG.md on potential PR approval)
```
Comment (hkaj) [inline libraries/datadogstatsd.php]: this PR will likely be accepted, might be a good time to move these :)

**Comment by hkaj** [comment]
A few nitpicks but it looks good, thanks again @folkhack! Once the comments are addressed we'll release this.

File: libraries/datadogstatsd.php (hunk 2 above)
Comment (zhukovra) [inline libraries/datadogstatsd.php]: Anyway, `ext-curl` or another requirement like `guzzle` must be in composer.json.

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,3 @@
 .project
 .settings
+.idea
```
Comment (zhukovra) [inline .gitignore]: Maybe new line?

**Comment by hush-hush** [comment]
This was merged with #41.

**Pull request closed without merging**